src/Compiler/FSComp.txt

```diff
@@ -1670,4 +1670,5 @@ featureEscapeBracesInFormattableString,"Escapes curly braces before calling Form
 3553,chkAdditionalConstructorOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Additional constructor is not allowed."
 3554,chkInstanceMemberOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Instance members are not allowed."
 3555,chkInstanceLetBindingOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Instance let bindings are not allowed."
-3556,chkImplementingInterfacesOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Implementing interfaces is not allowed."
\ No newline at end of file
+3556,chkImplementingInterfacesOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Implementing interfaces is not allowed."
```

Comment:
Implementing interfaces is not allowed except for interfaces with only 'static abstract' methods.

Reply:
Like this.

```
module Playground

[<Interface>]
type InputRetriever<'T when 'T:>InputRetriever<'T>> =
    static abstract Read: unit -> string

[<AbstractClass;Sealed>]
type ConsoleRetriever = 
    interface InputRetriever<ConsoleRetriever> with
        static member Read() = 
            stdout.WriteLine("Please enter a value and press enter")
            stdin.ReadLine()

let readInt<'T when InputRetriever<'T>>() = 'T.Read() |> int

[<EntryPoint>]
let main args = 
    let x = readInt<ConsoleRetriever>() + readInt<ConsoleRetriever>()
    x
```

Reply:
@T-Gro  Yeah. It begs the question if we should restrict this usage in static classes as they are not part of the spec. People can use a normal type/ class to do this? cc @vzarytovskii 

Happy update the condition to cover this if that is what we want :)

Reply:
> @T-Gro  Yeah. It begs the question if we should restrict this usage in static classes as they are not part of the spec. People can use a normal type/ class to do this? cc @vzarytovskii 
> 

One should be able to implement static abstracts in both static and non-static types.

Reply:
Updated following your suggestion. Thanks for the review :)